javaslang/src/main/java/javaslang/collection/PriorityQueue.java

```diff
@@ -0,0 +1,853 @@
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public Tuple2<? extends LinearSeq<T>, ? extends LinearSeq<T>> span(Predicate<? super T> predicate) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public LinearSeq<T> subSequence(int beginIndex) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public LinearSeq<T> subSequence(int beginIndex, int endIndex) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public LinearSeq<T> tail() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public Option<? extends LinearSeq<T>> tailOption() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public LinearSeq<T> take(long n) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public LinearSeq<T> takeRight(long n) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public LinearSeq<T> takeUntil(Predicate<? super T> predicate) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public LinearSeq<T> takeWhile(Predicate<? super T> predicate) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public <U> LinearSeq<U> unit(Iterable<? extends U> iterable) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public <T1, T2> Tuple2<? extends LinearSeq<T1>, ? extends LinearSeq<T2>> unzip(Function<? super T, Tuple2<? extends T1, ? extends T2>> unzipper) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public <T1, T2, T3> Tuple3<? extends Seq<T1>, ? extends Seq<T2>, ? extends Seq<T3>> unzip3(Function<? super T, Tuple3<? extends T1, ? extends T2, ? extends T3>> unzipper) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public LinearSeq<T> update(int index, T element) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public <U> LinearSeq<Tuple2<T, U>> zip(Iterable<? extends U> that) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public <U> LinearSeq<Tuple2<T, U>> zipAll(Iterable<? extends U> that, T thisElem, U thatElem) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public LinearSeq<Tuple2<T, Long>> zipWithIndex() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (o == this) {
+            return true;
+        } else if (o instanceof PriorityQueue) {
+            final PriorityQueue<?> that = (PriorityQueue<?>) o;
+            return this.toList().equals(that.toList());
+        } else {
+            return false;
+        }
+    }
+
+    @Override
+    public int hashCode() {
+        return toList().hashCode();
```

Comment:
> ... _IF_ ...

Exactly that's my point. Every List/Stream/Queue... node has a Lazy instance (including one Supplier + initially null value) and most probably it will never be used. I think that is the reason Scala also does not have it. We significantly reduced the HashMap in-memory size by removing all these Lazy instances.

I don't want to talk to Java guy and want to explain why our HashMap takes 200 MB space while Java's HashMap takes 30 MB when the caches are never called...